# Лабораторная работа 3.2. Домашнее задание

### Задание №1

Создайте два массива: в первом должны быть четные числа от 2 до 12 включительно, а в другом числа 7, 11, 15, 18, 23, 29.

$1.$ Сложите массивы и возведите элементы получившегося массива в квадрат:

In [2]:
import numpy as np

In [12]:
a = np.arange(2, 13, 2)
b = np.array([7, 11, 15, 18, 23, 29])

print("a = ", a)
print("b = ", b)
print((a + b) ** 2)

a =  [ 2  4  6  8 10 12]
b =  [ 7 11 15 18 23 29]
[  81  225  441  676 1089 1681]


$2.$ Выведите все элементы из первого массива, индексы которых соответствуют индексам тех элементов второго массива, которые больше 12 и дают остаток 3 при делении на 5.

In [13]:
print("a = ", a)
print("b = ", b)

print(a[(b > 12) & (b % 5 == 3)])

a =  [ 2  4  6  8 10 12]
b =  [ 7 11 15 18 23 29]
[ 8 10]


*3.* Проверьте условие "Элементы первого массива делятся на 4, элементы второго массива меньше 14". (Подсказка: в результате должен получиться массив с True и False)

In [16]:
print("Первое условие: ", a % 4 == 0)
print("Первое условие: ", b < 14)

Первое условие:  [False  True False  True False  True]
Первое условие:  [ True  True False False False False]


### Задание №2

* Найдите интересный для вас датасет. Например, можно выбрать датасет тут. http://data.un.org/Explorer.aspx (выбираете датасет, жмете на view data, потом download, выбирайте csv формат)
* Рассчитайте подходящие описательные статистики для признаков объектов в выбранном датасете
* Проанализируйте и прокомментируйте содержательно получившиеся результаты
* Все комментарии оформляйте строго в ячейках формата markdown
    

In [18]:
import csv
import numpy as np

In [34]:
with open('data.csv', 'r', newline='') as csvf:
    data = csv.reader(csvf, delimiter=';')
    values = []
    years = []
    for row in data:
        if row[0] == "Country or Area":
            continue
        values.append(float(row[5]))
        years.append(int(row[2]))
        
values_arr = np.array(values)
years_arr = np.array(years)

In [41]:
print(f"Среднее значение количества женщин к 100 мужчинам: {round(np.mean(values_arr),2)}")
print(f"Средний год рождения: {round(np.mean(years_arr),2)}")

print(f"\nМаксимальное значение отношения: {np.max(values_arr)}")
print(f"Максимальное значение года: {np.max(years_arr)}")

print(f"\nМинимальное значение отношения: {np.min(values_arr)}")
print(f"Минимальное значение года: {np.min(years_arr)}")

print(f"\nСреднее отклонение отношения: {round(np.std(values_arr),2)}")
print(f"Среднее отклонение года: {round(np.std(years_arr),2)}")

Среднее значение количества женщин к 100 мужчинам: 99.34
Средний год рождения: 1995.46

Максимальное значение отношения: 200.6
Максимальное значение года: 2004

Минимальное значение отношения: 69.0
Минимальное значение года: 1985

Среднее отклонение отношения: 9.23
Среднее отклонение года: 5.65


1) Получается, что выборка была из людей 1985-2004 года рождения (на 2023 год 19-38 лет).

2)Наибольшее отношение девушек к мужчинам 2 к 1

3) В среднем отклонение отношения составляет 9.23